src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const numbers: number[] = Array.from([a, b, c]).sort(
+      (numA: number, numB: number) => {
+        return numB - numA;
+      },
+    );
+
+    if (
+      numbers.some((num: number) => num <= 0) ||
+      numbers[0] >= numbers[1] + numbers[2]
+    ) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
The error message is well-detailed, which is good practice. No changes needed here.